client/src/components/signout-modal/index.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { Button, Modal } from '@freecodecamp/react-bootstrap';
+import { bindActionCreators, Dispatch, AnyAction } from 'redux';
+import { createSelector } from 'reselect';
+import { connect } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+
+import { ButtonSpacer } from '../helpers';
+import { hardGoTo as navigate, closeSignoutModal } from '../../redux/actions';
+import { isSignoutModalOpenSelector } from '../../redux/selectors';
+import { apiLocation } from '../../../../config/env.json';
+
+import './signout-modal.css';
+
+const mapStateToProps = createSelector(
+  isSignoutModalOpenSelector,
+  (show: boolean) => ({
+    show
+  })
+);
+
+const mapDispatchToProps = (dispatch: Dispatch<AnyAction>) =>
+  bindActionCreators(
+    {
+      navigate,
+      closeSignoutModal
+    },
+    dispatch
+  );
+
+type SignoutModalProps = {
+  navigate: (path: string) => void;
+  closeSignoutModal: () => void;
+  show: boolean;
+};
+
+function SignoutModal(props: SignoutModalProps): JSX.Element {
+  const { show, closeSignoutModal, navigate } = props;
+  const { t } = useTranslation();
+
+  const handleModalHide = () => {
+    closeSignoutModal();
+  };
+
+  const handleSignout = () => {
+    closeSignoutModal();
+    navigate(`${apiLocation}/signout`);
+  };
```

Comment:
Can you move the function back to nav-links file, and use prop instead? 

It's easier to find logic or function to fix or update, when all of them are in the same area, well it's hard to find that area sometimes, but if me from while found it, I think we are fine.

Reply:
Understood. 

But I think the logic should be declarative and follow functional design patterns. 

To elaborate, the Sign Out modal is a generic and idempotent component. It is, or its logic is NOT part of the nav (modal is a dependency of nav, not the other way).

You could trigger its "functionality" from anywhere, not just the nav.

That said, if code organization is a problem, I am all ears for something that doesn't break paradigms.

Reply:
My thoughts are this, this part is a section of the menu, so Ideally when we move the menu and its sections away from the nav-links, we have the logic in one place, it will be easier to change and move it to different section, when I try to add typescript back to the file.

Reply:
> have the logic in one place

Exactly what we need not be doing, not at the cost of breaking programming paradigms. I would recommend looking up functional design patterns.

Reply:
Ya, when I add typescript, I will move menu logic soon after. Just want to add type safety first before manipulating that file